content/client-lib-development-guide/features.textile

```diff
@@ -349,24 +386,25 @@ h3(#realtime-presence). Presence
 ** @(RTP7a)@ Unsubscribe with no arguments unsubscribes the listener if previously subscribed with an action-specific subscription
 ** @(RTP7b)@ Unsubscribe with a single action argument unsubscribes the provided listener to all presence messages for that action
 * @(RTP8)@ @Presence#enter@ function:
-** @(RTP8a)@ Enters this client into this channel
+** @(RTP8a)@ Enters the current client into this channel, optionally with the data provided
 ** @(RTP8b)@ Optionally a callback can be provided that is called for both success or failure to enter
-** @(RTP8c)@ A @PRESENCE ProtocolMessage@ with a @PresenceMessage@ with the action @ENTER@ is sent to the Ably service
+** @(RTP8c)@ A @PRESENCE ProtocolMessage@ with a @PresenceMessage@ with the action @ENTER@ is sent to the Ably service. The @clientId@ attribute of the @PresenceMessage@ must not be present. Entering without an explicit @PresenceMessage#clientId@, implicitly uses the @clientId@ for the current connection
 ** @(RTP8d)@ Implicitly attaches to the channel if not attached
 ** @(RTP8e)@ Optional data can be included when entering a channel that will be encoded / decoded as with normal messages. A test should exist to ensure data used with enter is encoded & decoded correctly. Also, when data is provided when entering, but no data is provided when leaving, the data attribute should be emitted in the @LEAVE@ event for this client
-** @(RTP8f)@ Raises an exception of the client library has not @clientId@ configured
+** @(RTP8f)@ Raises an exception if the client library does not have a @clientId@ configured and basic auth is being used
```

Comment:
The reason for this is that if you connect to Ably using an API key, and don't specify a `clientId` or any other argument that would make the client library use token authentication, then the client will be connected to Ably without having an implicit clientId.  If you then `enter`, the current client is not able to do that because `enter` requires the currently connected `client` to have an implicit or explicit `clientId`.  So the client library can raise an exception immediately as it will know there is no way this can work.  In the case of `RTP8i`, it is not the same as you can `enter` before the client is authenticated, and as such, we cannot raise an exception immediately.  We can only fail the `enter`, and emit an error explaining why this has happened.  I will try and reword this to make this all a bit clearer.